server/src/main/java/org/elasticsearch/index/search/stats/SearchStats.java

```diff
@@ -316,6 +338,8 @@ public void add(SearchStats searchStats) {
                 groupStats.get(entry.getKey()).add(entry.getValue());
             }
         }
+        isSearchIdle = searchStats.isSearchIdle;
```

Comment:
If shards stats are collected in index or data stream level, this could hide shards that are search idle.
Maybe we should report sum or max idle time? Or maybe we should move this information back to `ShardStats`?